### PR TITLE
Upgrade to Fabric v2.1.0

### LINKED ISSUE #347: Upgrade to Fabric 2.1
**Is your feature request related to a problem? Please describe.**
Assess whether changes from Fabric 2.0.1 to 2.1 affect any FPC code/protocol/design aspects and upgrade build, config files and scripts to rely on fabric 2.1

**Additional context**
[Release Notes Fabric v2.1](https://github.com/hyperledger/fabric/releases/tag/v2.1.0)

### PR DESCRIPTION
**Pull request by g2flyer**
**What this PR does / why we need it**:

Upgrades to the latest Fabric version, related dependencies and config files.

**Which issue(s) this PR fixes**:

Fixes #347 

**Special notes for your reviewer**:

To test you will have to move fabric from `v2.0.1` tag to `v2.1.0` tag

**Does this PR introduce a user-facing changes and/or breaks backward compatability?**:

Docu has changed correspondingly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -244,7 +244,7 @@ Make sure that you have the following required dependencies installed:
 
 * CMake v3.5.1 or higher
 
-* [Go](https://golang.org/) 1.14.x or higher
+* [Go](https://golang.org/) 1.14.1 or higher
```
Comment (MHBauer) [inline README.md:247]: is there a specific 1.4.1 feature, or that it matches fabric has published in 2.1.0 ?

File: integration/config/core.yaml [hunk 2]
```diff
@@ -382,8 +394,6 @@ peer:
     #   escc:
     #     name: DefaultESCC
     #     library: /etc/hyperledger/fabric/plugin/escc.so
-    #
-    #TODO: Enable FPC Additions once tlcc and plugins are intergrated
```
Comment (MHBauer) [inline integration/config/core.yaml:386]: todo complete?

File: integration/config/core.yaml [hunk 3]
```diff
@@ -401,6 +411,7 @@ peer:
           escc:
             name: DefaultEndorsement
             library:
+        #TODO: Enable below FPC Additions once validation plugins are re-intergrated
```
Comment (MHBauer) [inline integration/config/core.yaml:414]: no, moved here?

File: integration/config/core.yaml.orig [hunk 4]
```diff
@@ -1,746 +0,0 @@
-# Copyright IBM Corp. All Rights Reserved.
-# Copyright Intel Corp. All Rights Reserved.
```
Comment (MHBauer) [inline integration/config/core.yaml.orig:2]: file not used?

File: integration/config/orderer.yaml [hunk 5]
```diff
@@ -147,6 +146,7 @@ General:
         # client's time as specified in a client request message
         TimeWindow: 15m
 
+
```
Comment (MHBauer) [inline integration/config/orderer.yaml:149]: extra  lines?

File: integration/config/core.yaml [hunk 6]
```diff
@@ -3,7 +3,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 #
-# Based on sampleconfig from github.com/hyperledger/fabric.git tagged with v2.0.1 
+# Based on sampleconfig from github.com/hyperledger/fabric.git tagged with v2.1.0
```
Comment (MHBauer) [inline integration/config/core.yaml:6]: Do other config at same dir with same comment need to be updated?

File: README.md (hunk 1 above)
Comment (g2flyer) [reply README.md:247]: I just used 1.14.1 as this is what fabric explicitly called out,

File: integration/config/core.yaml (hunk 2 above)
Comment (g2flyer) [reply integration/config/core.yaml:386]: Nope, only part of it (tlcc) and  moved rest further down & closer to where the corresponding additions to be uncommented are

File: integration/config/core.yaml (hunk 3 above)
Comment (g2flyer) [reply integration/config/core.yaml:414]: yep

File: integration/config/core.yaml.orig (hunk 4 above)
Comment (g2flyer) [reply integration/config/core.yaml.orig:2]: this is not used and got in only by accident in an earlier PR

File: integration/config/orderer.yaml (hunk 5 above)
Comment (g2flyer) [reply integration/config/orderer.yaml:149]: this was extra space compared to the "template" from sampleconfig (i.e., when you do a `diff -r -u ../../hyperledger/fabric/sampleconfig integration/config`)

File: integration/config/core.yaml (hunk 6 above)
Comment (g2flyer) [reply integration/config/core.yaml:6]: good catch: i thought i grepped for all files containing 2.0.1 but evidently screwed up :-)

Comment (g2flyer) [reply integration/config/core.yaml:6]: there is now a corresponding fixup

File: fabric/README.md [hunk 7]
```diff
@@ -34,4 +34,4 @@ Aborting! Tag on current HEAD () does not match expected tag/v2.1.0!
 ```
 
 Seems that your Fabric is on the wrong branch.
-Try to run `pushd $FABRIC_PATH; git checkout tags/v2.0.1; popd;` followed by `make` again.
+Try to run `pushd $FABRIC_PATH; git checkout tags/v2.1.0; popd;` followed by `make` again.
```
Comment (MHBauer) [inline fabric/README.md:37]: not this one.

Comment (MHBauer) [reply fabric/README.md:37]: nope, was reading this as checkout for the cr

File: integration/config/core.yaml (hunk 6 above)
Comment (MHBauer) [reply integration/config/core.yaml:6]: not screw up, things happen, thanks for extra fixup.

**Comment by mbrandenburger** [comment]
Thanks!

**Pull request merged**